### ISSUE TITLE
InvalidResponseCodeException: Response code: 503 audio streaming

### ISSUE THREAD

**Issue by nikacotAndroid** [google/ExoPlayer#2142]
I am using the library for audio streaming only, using OkHttpDataSourceFactory and I am facing with issue on Samsung Galaxy S5 Android Version 5.0. The stream (http://88.85.99.43:8000/hitportal) is not working only with Mobile Data (3g/4g) and I am getting the following error:

```
E/ExoPlayerImplInternal: Source error.
                                                                           com.google.android.exoplayer2.upstream.HttpDataSource$InvalidResponseCodeException: Response code: 503
                                                                               at com.google.android.exoplayer2.upstream.DefaultHttpDataSource.open(DefaultHttpDataSource.java:212)
                                                                               at com.google.android.exoplayer2.upstream.DefaultDataSource.open(DefaultDataSource.java:123)
                                                                               at com.google.android.exoplayer2.source.ExtractorMediaPeriod$ExtractingLoadable.load(ExtractorMediaPeriod.java:596)
                                                                               at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:295)
                                                                               at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
                                                                               at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                                               at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
                                                                               at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
                                                                               at java.lang.Thread.run(Thread.java:818)
```

Any ideas how to fix the issue?

Thanks.

**Comment by ojw28** [comment]
The sample stream works fine for me over both WiFi and mobile data. It's unlikely to be an ExoPlayer issue. What you're seeing is a [http 503 response status code](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#5xx_Server_Error), which suggests something about your mobile connection is causing the issue. Are you sure there aren't any restrictions on mobile data enabled on the device (e.g. no background data + you're running your app in the background). In such cases I imagine the platform could choose to simulate a 503 status response as a way of indicating that connectivity isn't available.

As an aside, it doesn't look like you're using `OkHttpDataSourceFactory` to me. The stack trace shows `DefaultHttpDataSource` is being used.

**Comment by nikacotAndroid** [comment]
Oh yes, I posted wrong stack trace, since I was testing with both `OkHttpDataSourceFactory` and `DefaultHttpDataSource`, but still is the same error. 
It is really strange issue, since is happening just on this device with Mobile Data on. I tried to make hotspot from the device and connected to another, and on the other device the stream was working.
Definitely is something with the mobile data.
